### DISCUSSION TITLE
Metrics status?

### DISCUSSION THREAD

**Discussion by javaccar** [open-telemetry/opentelemetry-ruby#1371] [Q&A] [answered]
How can I track the status of the metrics implementation? For example, the opentelemetry-cpp project has been using project milestones (https://github.com/open-telemetry/opentelemetry-cpp/milestones) to share their progress, and I'm wondering if opentelemetry-ruby has something like this or if it's just a matter of skimming through PRs to see what has been accomplished and what's in progress. Also, does the opentelemetry-ruby project have a goal for when the metrics implementation will be released? Thanks

**Comment by ahayworth** [comment, marked as answer]
Hi @javaccar - right now, this is where we're tracking metrics progress right now: https://github.com/open-telemetry/opentelemetry-ruby/projects/2

We don't have a hard timeline for when it will be available, but tracking that project is the best way to stay up-to-date for now.

**Reply by javaccar** [reply]
thank you!

**Reply by clarissalimab** [reply]
Greetings, @ahayworth!

I'm a Ruby developer at @thoughtbot. My company will pay for my time contributing to OpenTelemetry Ruby, I'd be happy to help with the implementation of the metrics.

Could you help me with how I can get started with this? =)

**Reply by ahayworth** [reply]
Hi @clarissalimab - I'm extraordinarily pleased to hear that thoughtbot is willing to donate your time! Truthfully, we've been having a little trouble finding the motivation to get going on things (I'm sure partially because the holidays are approaching). So your contributions are most welcome, and well-timed!

For starters, we should make sure we actually have good issues (or at least project board "cards") for metrics: https://github.com/open-telemetry/opentelemetry-ruby/projects/2

I'm wondering if we couldn't split up the work into smaller chunks, which might make it a lot easier to work on and parallelize. I think this would also involve auditing the spec and making sure we have everything covered. If you're not familiar with the metrics spec, that might be a _very_ good place to start - it would accomplish two goals at once! 😄 🎉 

Otherwise, I think the next thing we should do is to get an MVP finalized with OTLP export:
- no async metrics, no views, etc
- only OTLP/HTTP export (there is some initial work for OTLP/GRPC but we shouldn't conflate the two)

Once we have that up and running in a way we're willing to publish to rubygems, we can start splitting the work up in earnest: multiple exporters, views, etc.

What do you think? Please feel free to ping me on the [cncf slack](https://slack.cncf.io) if you would prefer to chat a little more synchronously as well (my GitHub notifications are a dumpster fire right now).
